pkg/machine/provider/platform_windows.go

```diff
@@ -76,6 +72,22 @@ func InstalledProviders() ([]define.VMType, error) {
 	return installed, nil
 }
 
+func IsInstalled(provider define.VMType) (bool, error) {
+	switch provider {
+	case define.WSLVirt:
+		return wutil.IsWSLInstalled(), nil
+	case define.HyperVVirt:
+		service, err := hypervctl.NewLocalHyperVService()
+		service.Close()
```

Comment:
this is another nil deref on error, it can only be called on the err == nil case

Reply:
Updated, thanks!